Okta/OktaOidcHTTPProtocol.h

```diff
@@ -0,0 +1,16 @@
+//
+//  OktaOidcHTTPProtocol.h
+//  okta-oidc
+//
+//  Created by Lihao Li on 8/27/20.
+//  Copyright © 2020 Okta. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+
+@protocol OktaOidcHTTPProtocol <NSObject>
+
+- (void)willSendRequest: (nullable NSURLRequest *)request;
```

Comment:
Add full header documentation

Reply:
Don't forget to update README

Reply:
Yes, we have another ticket in MC: https://oktainc.atlassian.net/browse/OKTA-326267